src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java

```diff
@@ -6384,7 +6385,18 @@ public void execute(Integer arg) {
       releaseOnDismiss.add(prefs.showInvisibles().bind(
             new CommandWithArg<Boolean>() {
                public void execute(Boolean arg) {
-                  docDisplay.setShowInvisibles(arg);
+                  // show whitespace characters in '.csv', '.tsv' files
+                  boolean showInvisibles = arg;
+                  FileSystemItem file = context.getActiveFile();
+                  if (file != null)
+                  {
+                     String ext = file.getExtension();
+                     if (".tsv".contentEquals(ext) || ".csv".contentEquals(ext))
+                     {
+                        showInvisibles = true;
+                     }
+                  }
+                  docDisplay.setShowInvisibles(showInvisibles);
```

Comment:
I think this is a very sensible default but that we should add an escape hatch for the minority of users who don't want to see unprintable characters. What would you think about a hidden UI pref that defaults to on, so that as a last resort someone who can't stand seeing whitespace characters could run an R snippet to return to the previous behavior?